packages/backend.ai-ui/src/components/BAIUserNodes.tsx

```diff
@@ -0,0 +1,315 @@
+  ...availableUserSorterKeys,
+  ...availableUserSorterKeys.map((key) => `-${key}` as const),
+] as const;
+
+const isEnableSorter = (key: string) => {
+  return _.includes(availableUserSorterKeys, key);
+};
+
+interface BAIUserNodesProps
+  extends Omit<
+    BAITableProps<UserNodeInList>,
+    'dataSource' | 'columns' | 'onChangeOrder'
+  > {
+  usersFrgmt: BAIUserNodesFragment$key;
+  customizeColumns?: (
+    baseColumns: BAIColumnType<UserNodeInList>[],
+  ) => BAIColumnType<UserNodeInList>[];
+  disableSorter?: boolean;
+  onChangeOrder?: (
+    order: (typeof availableUserSorterValues)[number] | null,
+  ) => void;
+}
+
+const BAIUserNodes: React.FC<BAIUserNodesProps> = ({
+  usersFrgmt,
+  customizeColumns,
+  disableSorter,
+  onChangeOrder,
+  ...tableProps
+}) => {
+  'use memo';
+  const { t } = useTranslation();
+  const baiClient = useConnectedBAIClient();
+
+  const users = useFragment(
+    graphql`
+      fragment BAIUserNodesFragment on UserNode @relay(plural: true) {
+        id @required(action: NONE)
+        email @required(action: NONE)
+        full_name
+        role
+        description
+        username
+        created_at
+        modified_at
+        status
+        domain_name
+        resource_policy
+        allowed_client_ip
+        container_gids
+        container_main_gid
+        container_uid
+        status_info
+        sudo_session_enabled
+        need_password_change
+        totp_activated
+        project_nodes {
+          count
+          edges {
+            node {
+              name
+              id
+            }
+          }
+        }
+      }
+    `,
+    usersFrgmt,
+  );
+
+  const baseColumns = _.map(
+    filterOutEmpty<BAIColumnType<UserNodeInList>>([
+      {
+        key: 'email',
+        title: t('comp:UserNodes.Email'),
+        sorter: isEnableSorter('email'),
+        dataIndex: 'email',
+        fixed: true,
+        render: (__, record) => <BAIText copyable>{record.email}</BAIText>,
+        required: true,
+      },
+      {
+        key: 'id',
+        title: 'ID',
+        render: (__, record) => (
+          <BAIText
+            copyable
+            ellipsis
+            monospace
+            style={{
+              maxWidth: 100,
+            }}
+          >
+            {toLocalId(record.id)}
+          </BAIText>
+        ),
+      },
+      {
+        key: 'username',
+        title: t('comp:UserNodes.Username'),
+        dataIndex: 'username',
+        sorter: isEnableSorter('username'),
+      },
+      {
+        key: 'full_name',
+        title: t('comp:UserNodes.FullName'),
+        dataIndex: 'full_name',
+        sorter: isEnableSorter('full_name'),
+      },
+      {
+        key: 'domain_name',
+        title: t('comp:UserNodes.DomainName'),
+        dataIndex: 'domain_name',
+        minWidth: 100,
+        sorter: isEnableSorter('domain_name'),
+      },
+      baiClient?.isManagerVersionCompatibleWith?.('25.15.2') && {
```

Comment:
This part should also be managed in ESM and verified using `baiClient.supports`.